applications/CompressiblePotentialFlowApplication/custom_elements/compressible_perturbation_potential_flow_element.cpp

```diff
@@ -472,8 +472,8 @@ void CompressiblePerturbationPotentialFlowElement<Dim, NumNodes>::CalculateLeftH
     CalculateLeftHandSideContribution(lower_lhs_total, rCurrentProcessInfo, lower_velocity, data);
 
     // Compute lhs wake condition
-    const double free_stream_density = rCurrentProcessInfo[FREE_STREAM_DENSITY];
-    const BoundedMatrix<double, NumNodes, NumNodes> lhs_wake_condition = data.vol * free_stream_density * prod(data.DN_DX, trans(data.DN_DX));
+    const BoundedMatrix<double, NumNodes, NumNodes> lhs_wake_condition =
+        CalculateLeftHandSideWakeConditions(data, rCurrentProcessInfo);
```

Comment:
@marcnunezc, @philbucher, @rubenzorrilla would you suggest using reference here too since it is anyway const?

```suggestion
    const BoundedMatrix<double, NumNodes, NumNodes>& lhs_wake_condition =
        CalculateLeftHandSideWakeConditions(data, rCurrentProcessInfo);
```

Reply:
After asking this to @roigcarlo (please correct or edit my message if I say something wrong), the conclusion is that the reference is not needed in this case. 
The compiler will perform "Return Value Opimization" (see https://en.cppreference.com/w/cpp/language/copy_elision):

> In the initialization of an object, when the source object is a nameless temporary and is of the same class type (ignoring cv-qualification) as the target object. When the nameless temporary is the operand of a return statement, this variant of copy elision is known as RVO, "return value optimization".

Ensuring that the assignment is performed efficiently

However, in general terms, one need to be careful when using references in these cases. In the example (author @roigcarlo):

```
int& foo(int *a) {
    return *a;
}

int& boo() {
    int a = 0;
    return a
}
```
In `foo()`, there would be no issue since `a` exists out of the scope of the function. But in `boo()`, `a` would be deleted after the function usage, giving an invalid address.

In our case the return statement is a temporary operation:
https://github.com/KratosMultiphysics/Kratos/blob/c6b0f161c55e42e9cb0646e9795d306df774ea5d/applications/CompressiblePotentialFlowApplication/custom_elements/compressible_perturbation_potential_flow_element.cpp#L521
 so it different than the example above, and RVO happens

Reply:
Ok! I'll try to remember this for future PRs :+1: